server/store/sqlstore/store.go

```diff
@@ -936,8 +936,8 @@ func (s *SQLStore) IsUserPresentInWhitelist(userID string) (bool, error) {
 	return result != "", nil
 }
 
-func (s *SQLStore) ListDMsGMsToConnect() ([]string, error) {
-	query := s.getQueryBuilder().Select("c.Id").From("Channels AS c").LeftJoin("sharedchannelremotes AS scr ON scr.channelid = c.id").Where(sq.Eq{"scr.remoteid": nil}).Where(sq.Or{sq.Eq{"c.Type": "G"}, sq.Eq{"c.Type": "D"}})
+func (s *SQLStore) ListDMsGMsToConnectBatch() ([]string, error) {
+	query := s.getQueryBuilder().Select("c.Id").From("Channels AS c").LeftJoin("sharedchannelremotes AS scr ON scr.channelid = c.id").Where(sq.Eq{"scr.remoteid": nil}).Where(sq.Or{sq.Eq{"c.Type": "G"}, sq.Eq{"c.Type": "D"}}).Limit(500)
```

Comment:
To confirm, this query doesn't use our own `p.remoteId` at all: is that because a shared channel is shared "once" for all remotes? What does the `scr.remoteid` mean in that case?

Reply:
Good catch, this works because we don't have DMs connected to any other shared channel thing, so I was checking if there is not remoteId, what means, the channel is not connected, I think the question is more "remoteId != p.remoteId", let me change that.